### ISSUE TITLE
How would one handle runtime reconfiguration?

### ISSUE THREAD

**Issue by tuboythers** [tokio-rs/tokio#2111]
Hello, this is not much of an issue, rather a general question since I've banged my head in the wall for a few days now and cannot really come up with viable solution. So the simple question is how would one handle reload of lets say some data from a file once the future has been spawned? 

Sample code:

```
tokio::spawn(async move {
   /// Execute a function which reads data from a file
   /// Spawn a TcpListener and process streams according to that data
});

tokio::spawn(async move {
    /// Here create a Rusls ServerConfig
    /// Start accepting TLS Encrypted connections
});
````

So, say I have these two tasks in a single main function of my code and each of these read some configuration, the first task from a file, the second task is Rustls based one. If you were on my place and wanted to for example, to make some change in the configuration file and update the application without restarting/quitting it, how would you've handled such case? 

I am aware this could be related to graceful restart/shutdown cases, but I really cannot get my mind around this and would appreciate any help on the matter. Apologies for not structuring this according to the Issues template.

**Comment by LucioFranco** [comment]
I would probably treat the system as restartable, meaning that I can gracefully shutdown my spawned task and then restart it without issue and pretty quickly. Then you can use a oneshot channel to send a signal coupled with a `select` you can then gracefully start shutting down. This may mean draining all tcpstreams etc. Then once the shutdown is complete you could then respawn a new task with the new config.

Here is a big of a higher level explanation of more general concepts like this https://github.com/tokio-rs/tokio/issues/1879

Hope that helps!

**Comment by Darksonn** [comment]
This is the purpose of the [`watch`][1] channel.

[1]: https://docs.rs/tokio/0.2/tokio/sync/watch/index.html
